### ISSUE TITLE
Missing event for `doOpenAuth` failure

### ISSUE THREAD

**Issue by jjenzz** [hirosystems/connect#186]
I'm using `react-connect` but seems applicable to `connect` also.

When calling `doOpenAuth` in an incognito window, the hiro wallet window doesn't open but my app gets stuck in "loading" state. It would be great if there was a way to detect a failure so I can communicate that in the UI.

Video here: https://twitter.com/jjenzz/status/1485244796470009858?s=20

**Comment by fbwoolf** [comment]
I agree, this would be helpful. Are you getting any feedback in the console as to why it is failing?

**Comment by jjenzz** [comment]
Looks like something to do with the `connect-ui` package:

![image](https://user-images.githubusercontent.com/175330/150784474-b996cb57-e8c6-4a06-8c50-f587a0804541.png)

I'm using https://remix.run which uses esbuild if that helps at all.

**Comment by He1DAr** [comment]
This issue is caused by `esbuild` [not supporting dynamic imports via plugins](https://github.com/evanw/esbuild/issues/700). When `Connect` fails to find a wallet (e.g. in incognito if wallet plugin is not enabled in incognito mode), it attempts to open the modal:
![image](https://user-images.githubusercontent.com/97111756/150893624-e13594db-04c7-48b8-8d1d-7d0d90563026.png)
and fails due to `esbuild` lack of dynamic imports support.

**Comment by jjenzz** [comment]
Thanks for looking into that, I assumed that would be the case though as I have had other issues with esbuild up to this point. 

The ticket is about providing an `onFailure` event of some kind though because there is no way for my app to know something like this happened currently (e.g. connection drops and dynamic import fails).

**Comment by He1DAr** [comment]
> Thanks for looking into that, I assumed something like that would be the case as I have had other issues with esbuild up to this point.
> 
> The ticket is about providing an `onFailure` event of some kind though because there is no way for my app to know something like this happened currently (e.g. connection drops and dynamic import fails).

@jjenzz have you tried wrapping your component with [error boundary](https://reactjs.org/docs/error-boundaries.html)?

**Comment by jjenzz** [comment]
I tried wrapping `doOpenAuth()` in a `try {} catch(e) {}` before which didn't catch it and have just tried an `ErrorBoundary` without success also.

**Comment by markmhx** [comment]
I'll review this issue with @sabbyanandan, @saralab and @janniks soon. 🙏
